backend/editor/entries.py

```diff
@@ -101,4 +186,40 @@ def update_nodes(label, entry, incomingData):
 
     params = dict(incomingData, id=entry)
     result = session.run(" ".join(query), params)
-    return result
\ No newline at end of file
+    return result
+
+def update_node_children(entry, incomingData):
+    """
+    Helper function used for updation of node children with given id 
+    """
+    # Parse node ids from Neo4j Record object
+    current_children = [record["b.id"] for record in list(get_children(entry))]
+    deleted_children = list(set(current_children) ^ set(incomingData))
+
+    # Delete relationships
+    for child in deleted_children:
+        query = []
+        main_language_code = child[:2]
+        is_old_node = list(get_nodes("ENTRY", child))

```

Comment:
It would be better to check existence of all nodes in only one request  and then again to get non_existing nodes by comparing new ids and ids in database.
It's not blocking though, you can put a TODO if you prefer.

Reply:
I separated the check for non-existing nodes, from the "creation of new relationships" part. Currently, the node creates a new code if it doesn't exist. After all nodes have been created, the relationships are built. Do you think I'd need to change anything here?

Reply:
We may keep it like it, my concern was about performance:

You could do:
```python
    # create non existing node
    existing = session.run("""MATCH (child:ENTRY) WHERE id in $ids RETURN n.id""", ids=added_children)
    to_create = added_children - set(existing)
    for child in to_create:
        main_language_code = child.split(":", 1)[0]
        create_node("ENTRY", child, main_language_code)
    # create relations
    for child in added_children:
    ...
```